packages/pretty-format/src/index.js

```diff
@@ -191,14 +193,15 @@ function printMap(val, indent, prevIndent, spacing, edgeSpacing, refs, maxDepth,
   return result + '}';
 }
 
-function printObject(val, indent, prevIndent, spacing, edgeSpacing, refs, maxDepth, currentDepth, plugins, min, callToJSON, printFunctionName, escapeRegex, colors) {
+function printObject(val: Object, indent: string, prevIndent: string, spacing: string, edgeSpacing: string, refs: Refs, maxDepth: number, currentDepth: number, plugins: Plugins, min: boolean, callToJSON: boolean, printFunctionName: boolean, escapeRegex: boolean, colors: Colors): string {
   const constructor = min ? '' : (val.constructor ?  val.constructor.name + ' ' : 'Object ');
   let result = constructor + '{';
   let keys = Object.keys(val).sort();
   const symbols = getSymbols(val);
 
   if (symbols.length) {
     keys = keys
+      // $FlowFixMe string literal `symbol`. This value is not a valid `typeof` return value
       .filter(key => !(typeof key === 'symbol' || toString.call(key) === '[object Symbol]'))
```

Comment:
Aha, I thought I ran into this limitation of Flow before. It’s part of ES2015.

* https://github.com/facebook/flow/issues/1015
* http://eslint.org/docs/rules/valid-typeof

So the second part might be a fall-back? Notice:

```js
  const typeOf = typeof val;
  if (typeOf === 'symbol') {
    return printSymbol(val);
  }
```

Will wait for second option and adjust corresponding test to emphasize correct sorting :)